external-docs/docs/guides/configuration/qube-config/columns/units.md

```diff
@@ -0,0 +1,145 @@
+rows use `Inches` and the third row uses `Centimetres` to measure the `Average Height of Men`.
+
+## Basic configuration
+
+The following JSON shows how a units column can be defined in a [qube configuration file](../index.md):
+
+```json
+{
+    "$schema": "https://purl.org/csv-cubed/qube-config/v1",
+    "title": "Average Height of Men in different countries",
+    "columns": {
+        "Unit": {
+            "type": "units"
+        },
+    }
+}
+```
+
+To define a units column, specify the `type` of the column definition as `units`.
+
+<!-- TODO: At some point, add an example of a single measure pivoted data set with a units column (multiple units) -->
+
+todo: write about default behaviour of units column configuration
+
+## Optional properties
+
+When defining a units column, there are optional properties that can be entered, depending on how your units are
+being defined within the column.
+
+## values
+
+If you are creating new units within your units column, the details of the new units should be entered into a
+`values` field. The JSON below shows an example of the `values` field used in a units column.
+
+```json
+{
+    "$schema": "https://purl.org/csv-cubed/qube-config/v1",
+    "title": "Average Height and Weight for Men in different countries",
+    "columns": {
+        "Unit": {
+            "type": "units",
+            "values": true
+        },
+    }
+}
+```
+
+By default, the `values` field is set to `true`. This indicates to csvcubed to automatically generate
+[unit definitions](../unit-definitions.md) unique to your data set.
+
+## from_template
+
+Units columns can also make use of the `from_template` field. Templates are pre-configured column definitions that help
+speed up the creation of linked data columns. The use of this field in a units column is shown in the example below:
+
+```json
+{
+    "$schema": "https://purl.org/csv-cubed/qube-config/v1",
+    "title": "Average Income in Pounds Sterling",
+    "columns": {
+        "Unit": {
+            "from_template": "http://qudt.org/vocab/unit/PoundSterling"
+        },
+    }
+}
+```
+
+Enter the URI of a template to be used as the units column. Note that you can override the configuration of used
+templates by specifying individual properties.
+
+For more information on templates, as well as a list of templates that can be used, see the
+[Templates](../templates.md) page.
+
+## cell_uri_template
+
+If you are re-using existing units in your measures column, then do not use the `values` field to define the
+unit details. Instead, use the field `cell_uri_template`.
+
+```json
+{
+    "$schema": "https://purl.org/csv-cubed/qube-config/v1",
+    "title": "Average Height and Weight for Men in different countries",
+    "columns": {
+        "Unit": {
+            "type": "units",
+            "cell_uri_template": "http://example.org/units/example-unit"
```

Comment:
I'd expect to see a variable inside this `cell_uri_template` referring back to the csvw_name_safed column name.